File: bee/beginners/1049.py
'''
1049 - ANIMAL

Neste problema, você deverá ler 3 palavras que definem o tipo de animal possível segundo o esquema abaixo, da esquerda para a direita.  Em seguida conclua qual dos animais seguintes foi escolhido, através das três palavras fornecidas.

	VERTEBRADO = AVE > CARNIVORO = AGUIA
				     > ONIVORO = POMBA

			   = MAMIFERO > ONIVORO = HOMEM
			   			  > HERBIVORO = VACA

	INVERTEBRADO = INSETO > HEMATOFAGO = PULGA
						  > HERBIVORO = LAGARTA

			     = ANELIDEO > HEMATOFAGO = SANGUESSUGA
				 			> ONIVORO = MINHOCA

Entrada
A entrada contém 3 palavras, uma em cada linha, necessárias para identificar o animal segundo a figura acima, com todas as letras minúsculas.

Saída
Imprima o nome do animal correspondente à entrada fornecida.
'''
grupo = input()
classe = input()
habito = input()

if (grupo == 'vertebrado'):
	if(classe == 'ave'):
		if(habito == 'carnivoro'):
			print("aguia")
		else:
			print("pomba")
	else:
		if(habito == 'onivoro'):
			print("homem")
		else:
			print("vaca")

else:
	if(classe == 'inseto'):
		if(habito == 'hematofago'):
			print("pulga")
		else:
			print("lagarta")
	else:
		if(habito == 'hematofago'):
			print("sanguessuga")
		else:
			print("minhoca")

File: beecrowd/cpp/beginners/1049.cpp
#include <iostream>
#include <cctype>
#include <algorithm>
#include <string>

int main() {
    // Subfilo = Vertebrados e Invertebrados
    // Tipo = Ave, Mamifero, Inseto ou Anelideo
    // Ordem = Carnivoro, Onivoro, Herbivoro e Hematofago
    std::string subfilo, tipo, ordem;
    std::cin >> subfilo >> tipo >> ordem;

    // Validação de strings para lowercase
    std::transform(subfilo.begin(), subfilo.end(), subfilo.begin(), ::tolower);
    std::transform(tipo.begin(), tipo.end(), tipo.begin(), ::tolower);
    std::transform(ordem.begin(), ordem.end(), ordem.begin(), ::tolower);

    if (subfilo == "vertebrado") {
        if (tipo == "ave") {
            if (ordem == "carnivoro") {
                std::cout << "aguia" << std::endl;
            } else {
                std::cout << "pomba" << std::endl;
            }
        } else {
            if (ordem == "onivoro") {
                std::cout << "homem" << std::endl;
            } else {
                std::cout << "vaca" << std::endl;
            }
        }
    } else {
        if (tipo == "inseto") {
            if (ordem == "hematofago") {
                std::cout << "pulga" << std::endl;
            } else {
                std::cout << "lagarta" << std::endl;
            }
        } else {
            if (ordem == "hematofago") {
                std::cout << "sanguessuga" << std::endl;
            } else {
                std::cout << "minhoca" << std::endl;
            }
        }
    }
}